File: 20190606/hw/hw.php
<?php
function find($data, $find) {
    $left = 0;
    $right = sizeof($data) - 1;
    while ($left <= $right) {
        $mid = floor($left+$right);
        if ($data[$mid] == $find) {
            return $mid;
        } else if ($data[$mid] > $find) {
            $right = $mid - 1;
        } else {
            $left  = $mid + 1;
        }
    }
    return -1;
}

echo find(array(1, 2, 3), 2);

File: 20190609/hw/hw.cc
#include <iostream>

using namespace std;

int main() {
    char str[] = {'h', 'w', '\n'};
    for (int i = 0; i < sizeof(str)/sizeof(str[0]); i++) {
        printf("%c", str[i]);
    }
    return 0;
}
